packages/server-wallet/e2e-testing/scripts/sanity-checker.ts

```diff
@@ -0,0 +1,165 @@
+import yargs from 'yargs';
+import {hideBin} from 'yargs/helpers';
+import jsonfile from 'jsonfile';
+import chalk from 'chalk';
+import {BN} from '@statechannels/wallet-core';
+import {Uint256} from '@statechannels/nitro-protocol';
+import _ from 'lodash';
+import {BigNumberish, utils} from 'ethers';
+import columnify from 'columnify';
+
+import {Assertion, ChainState, RoleConfig, Step} from '../types';
+import {defaultTestWalletConfig, overwriteConfigWithDatabaseConnection} from '../../src';
+import {getKnexFromConfig} from '../../src/db-admin/db-admin';
+import {ObjectiveModel} from '../../src/models/objective';
+import {Channel} from '../../src/models/channel';
+import {setupUnhandledErrorListeners} from '../utils';
+
+setupUnhandledErrorListeners();
+assertSanity();
+
+async function assertSanity() {
```

Comment:
enhancement: I think checking the overall state of the user's balance is a good thing to check here, but can be added to the sanity checker on an as-needed basis.

For instance, if we end the simulation believing there are 34 active ledger-funded channels and 21 active virtually-funded channels, then we should somehow check that's reflected in the DB. This would be pretty close to taking a "snapshot" of the system mid-use, and making sure that everything is as we expect.

(I think there are a lot of ways of solving this general problem of "how do we know it's working correctlY?" and don't mean to express a strong opinion on _how_ to answer this question, just on its importance!)